### PR TITLE
:sparkles: Adding Calendar View

### PR DESCRIPTION
**Pull request by chriskryn**

### CODE REVIEW

File: teferi/ViewControllers/MainViewController.swift [hunk 1]
```diff
@@ -39,6 +39,25 @@ class MainViewController : UIViewController, MFMailComposeViewControllerDelegate
     private var addButton : AddTimeSlotView!
     private var permissionView : PermissionView?
     private var launchAnim : LaunchAnimationView!
+    private var _calendarViewController: CalendarViewController?
+    fileprivate var calendarViewController: CalendarViewController?
```
Comment (willsb) [inline teferi/ViewControllers/MainViewController.swift]: I think a `lazy` property would fit better here

File: teferi/ViewControllers/MainViewController.swift [hunk 2]
```diff
@@ -169,20 +188,114 @@ class MainViewController : UIViewController, MFMailComposeViewControllerDelegate
...
-            [ TimelineViewController(date: today,
-                                     metricsService: self.metricsService,
-                                     timeSlotService: self.timeSlotService,
-                                     editStateService: self.editStateService) ],
-            direction: .forward,
-            animated: true,
-            completion: nil)
+        if let calendarViewController = self.calendarViewController
+        {
+            if calendarViewController.view.alpha == 0
```
Comment (willsb) [inline teferi/ViewControllers/MainViewController.swift]: I don't like the alpha checking here. Ideally, we should either have an `isVisible` property on the `CalendarViewController` or simply have a `ToggleVisibility()` method that does all the work. I prefer the later as it'd remove responsability from the MainViewController, but I'd like to see @amulware's idea on this.

File: teferi/ViewControllers/MainViewController.swift [hunk 3]
```diff
@@ -169,20 +188,114 @@ class MainViewController : UIViewController, MFMailComposeViewControllerDelegate
...
+                default:
+                    break
+                }
+            })
+            .addDisposableTo(disposeBag!)
         
-        self.onDateChanged(date: today)
+    }
+    
+    func hideCalendar(calendarViewController: CalendarViewController,
```
Comment (willsb) [inline teferi/ViewControllers/MainViewController.swift]: 1 - Why is this public?
2 - Why receive an instance of the VC when you can simply access the field?

File: teferi/ViewModels/CalendarViewModel.swift [hunk 4]
```diff
@@ -0,0 +1,45 @@
+//
+//  CalendarViewModel.swift
+//  teferi
+//
+//  Created by Krzysztof Kryniecki on 11/29/16.
+//  Copyright © 2016 Toggl. All rights reserved.
+//
+
```
Comment (willsb) [inline teferi/ViewModels/CalendarViewModel.swift]: Remove this

File: teferi/Views/CalendarCellView.swift [hunk 5]
```diff
@@ -0,0 +1,125 @@
+//
+//  CalendarCellView.swift
+//  teferi
+//
+//  Created by Krzysztof Kryniecki on 11/28/16.
+//  Copyright © 2016 Toggl. All rights reserved.
+//
+
```
Comment (willsb) [inline teferi/Views/CalendarCellView.swift]: Remove this

File: teferi/Views/CalendarCellView.swift [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+import UIKit
+import JTAppleCalendar
+
+class CalendarCellView: JTAppleDayCellView
+{
+
+    @IBOutlet weak var dateLabel: UILabel!
+    @IBOutlet weak var activityView: UIView!
+    let lightGreyColor = UIColor(colorLiteralRed: 206.0/255.0,
```
Comment (willsb) [inline teferi/Views/CalendarCellView.swift]: This should be included in the `Colors` class

**Comment by amulware** [comment]
Tried checking out and building this, but it fails with:
```teferi/ViewControllers/CalendarViewController.swift:97:14: Value of type 'JTAppleCalendarView' has no member 'scrollToPreviousSegment'
teferi/ViewControllers/CalendarViewController.swift:102:14: Value of type 'JTAppleCalendarView' has no member 'scrollToNextSegment'```

**Comment by chriskryn** [comment]
Have you done 'pod install' before building?
I added JTAppleCalendar to app pods.

**Comment by chriskryn** [comment]
Ok I have found the problem, they must have released newer version of JTAppleCalendar and it doesn't have scrollToPreviousSegment and scrollToNextSegment. I have updated the podfile to use correct version.

**Comment by amulware** [comment]
Work on this will continue on our own branch.

**Pull request closed without merging**